torch/ao/quantization/backend_config/backend_config.py

```diff
@@ -63,32 +72,95 @@ class ObservationType(Enum):
     example: torch.cat, maxpool
     """
 
+
+@dataclass
+class DTypeWithConstraints:
+    """
+    Config for specifying additional constraints for a given dtype, such as quantization value
+    ranges and scale value ranges, to be used in :class:`~torch.ao.quantization.backend_config.DTypeConfig`.
+    """
+    dtype: torch.dtype
+    quant_min: Optional[float] = None
```

Comment:
should this be a float, or should it match the dtype of the tensor's storage?

Reply:
I just used float here for simplicity, not sure how to match the dtype of the tensor's storage without complicating the signatures

Reply:
Comparisons shouldn't involve casting logic between dtypes.  I'd vote for changing the signature to make sure comparisons are done without having to cast.

Reply:
Changed to `Optional[Union[int, float]]`